Document audit log streaming feature (#9556)
circleci/circleci-docs: docs/guides/modules/security/pages/audit-logs.adoc

= Audit logs

> :page-platform: Cloud

Reviewer: General things - sentence case for all headings and all things you are highlighting in bullet lists

NOTE: In some situations, the internal system may generate duplicate events in the audit logs. The `id` field of the downloaded logs is unique per event and can be used to identify duplicate entries.

Per link:https://circleci.com/privacy/#information[our data retention policy], audit logs can be retrieved for up to 12 months.

NOTE: Submitted dates are in UTC. To avoid issues arising from potential time zone differences, CircleCI adds extra time to the request. For example, if you request August 2 - August 5, the returned results will be in the range August 1 - August 6. The audit log request column is also displayed in UTC.

[#request-status]
=== Request status

In the link:https://app.circleci.com/[CircleCI web app], a status will show the following information:

- Timeframe requested
- User who made the request.
- Date request was made.
- Expiry date of the request.
- Request status.

Statuses are updated once per hour on the 30-minute mark (for example, 09:30, 10:30, 11:30).

Successful requests can be either of the following:

- Active with a download link.
- Active without any data (no download link).
- Expired (no longer available to download).

> Once your request is successfully processed, select btn:[Download logs] to download the audit log as a CSV file.

Reviewer: ```suggestion
Once your request is successfully processed, select btn:[Download logs] to download the audit log as a CSV file. Successful requests are available to download for 30 days.
```

NOTE: Audit log fields with nested data contain JSON blobs.

> NOTE: Successful requests are available to download for 30 days.

Reviewer: ```suggestion
```

image::guides:ROOT:audit-log-request.png[Audit log requests]

[#frequency-limits]
=== Frequency limits

NOTE: CircleCI may update the quantity and request period in the future.

Frequency limits for requests are as follows:

- Free Plan: 1 request per day
- All paid plans: 3 requests per day
- Maximum query window for 30 days - specifically, a customer can request logs for any 30 day window where the start of the window is within the last calendar year

If your organization has reached the maximum amount of requests per 30 days, the audit log request feature will be disabled.

If you hover over the disabled btn:[Request audit logs] button, a tooltip will appear displaying the date when new requests can be made. Pending requests count toward the rate limit.

[#audit-log-streaming]

> == Audit Log Streaming

Reviewer: As above, == Stream audit logs

NOTE: Even if you set up audit log streaming, you can still send xref:#audit-log-requests[audit logs requests] to CircleCI.

> The CircleCI Audit Log Streaming feature provides organizations with a solution for streaming audit logs directly to their own AWS S3 buckets.
> 
> This functionality enables organizations to maintain comprehensive security visibility, meet compliance requirements, and integrate CircleCI audit data with their existing security monitoring infrastructure.

Reviewer: ```suggestion
Set up audit log streaming direct to your own AWS S3 bucket.

Audit log streaming provides the following benefits to your organization: 

* Maintain comprehensive security visibility.
* Meet compliance requirements.
* Integrate CircleCI audit data with your existing security monitoring infrastructure.
```

[#prerequisites]
=== Prerequisites

[NOTE]
====
Before setting up audit log streaming, ensure you have the following:

* CircleCI Scale plan (not available to Free or Performance customers)
* AWS account with permissions to create and manage S3 buckets
* IAM role or user with appropriate S3 write permissions
* Network configuration to allow secure connections between CircleCI and AWS

> ====

Reviewer: This can just be normal text not a note admonition

> The following steps describe how to set up AWS prerequisites:

Reviewer: needs a title, for example === Set up AWS bucket and authentication (is that the right word?)

+
The minimum required access policy for the role is as follows:
+
[source,json]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:GetObject",
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::<bucket-name>/*",
                "arn:aws:s3:::<bucket-name>"
            ]
        }
    ]
}
----
+
Replace `<bucket-name>` with the name of your S3 bucket.

. Set up an OIDC identity provider to allow CircleCI's OIDC tokens to be used for authentication.
.. Navigate to the IAM Console in AWS.
.. In the left navigation menu, select *Identity Providers*.
.. Select *Add provider*.
.. Choose *OpenID Connect* as the provider type.
.. In the *Provider URL* field, specify `\https://oidc.circleci.com/org/<org-id>` (replace `<org-id>` with your CircleCI organization ID).
.. In the *Audience* field, enter your organization ID.

[#configure-audit-log-streaming]
=== Configure Audit Log Streaming

. Navigate to menu:Organization Settings[Security].

> . Scroll down to the **Audit Logs > Streaming audit logs** section and select btn:[Stream audit logs].

Reviewer: you can use the menu:[] macro throughout. The > looks a little different with and without
<img width="155" height="82" alt="Screenshot 2025-09-08 at 12 10 41" src="https://github.com/user-attachments/assets/30c4b1de-c21b-443c-8201-f6a44b9aea3b" />

+
image::guides:ROOT:stream-audit-logs.png[Stream audit logs section]

. Confirm that you have completed the prerequisites by selecting btn:[Continue].
+
image::guides:ROOT:confirm-prerequisites.png[Confirm prerequisites completion]

. In the configuration dialog, specify the following fields:
.. **Region**: The AWS region where your S3 bucket is located.
.. **S3 Bucket Name**: The name of the S3 bucket where audit logs will be streamed.
.. **Role ARN**: The Amazon Resource Name (ARN) for the IAM role that CircleCI will use to access your S3 bucket.

+
image::guides:ROOT:connect-circleci-aws.png[Connect CircleCI to AWS]

. Select btn:[Connect CircleCI to AWS].

> . When the connection to your S3 bucket is successfully validated, **Connected** with a green checkmark appears under **Status**.

Reviewer: we actually have the `passed` icon available for this

+
image::guides:ROOT:s3bucket-connected.png[S3 Connection successful]

[#verification-and-testing]
=== Verification and Testing

> After setting up audit log streaming:

Reviewer: ```suggestion
After setting up audit log streaming follow these steps to validate the setup was successful:
```

Reviewer: just a suggestion, whatever you think works

. Verify successful streaming by checking the **Last delivery** timestamp.
. Confirm logs are appearing in your designated S3 bucket.

> . Test integrations with your SIEM or security monitoring tools.

Reviewer: what is SIEM? Can you spell that out and have (SIEM) in brackets?

> . Set up appropriate alerts for connection failures or delivery interruptions.

Reviewer: Do we want to prescribe that people do this?

Author: I _think_ so. It would help catch issues early on.

[#managing-active-streams]
=== Managing Active Streams

> Once configured, you can view and manage audit log streams.

Reviewer: add something like, "The following options are available to you in the CircleCI web app"

- Pause Streaming: Use the toggle switch under **Actions** to temporarily disable log streaming. **Status** will change to **Disabled**.
- Monitor Delivery: Check the **Last delivery** field to verify recent successful deliveries.
- Delete Configuration: Select the trash icon under **Actions** to remove the streaming setup. A popup warning will appear, select btn:[Delete] to confirm.

> - Restart Streaming: Toggle the switch to resume paused streams.

Reviewer: sentence case
